visualdl/utils/logging.h

```diff
@@ -64,24 +64,22 @@ struct Error : public std::runtime_error {
   explicit Error(const std::string& s) : std::runtime_error(s) {}
 };
 
-// With exception.
+// With exception. The throw occurs in the destructor.
+// Please follow the pattern: LogStreamFatal(__FILE__, __LINE__).stream()
+// so the destructor will immediately execute.
 struct LogStreamFatal {
   LogStreamFatal(const char* file, int line) {
     ss << "[" << file << ":" << line << "] ";
-    throw Error(ss.str());
   }
 
   std::stringstream& stream() { return ss; }
 
   ~LogStreamFatal() {
-    if (!has_throw_) {
-      std::cerr << "throw exception" << std::endl;
-      throw Error(ss.str());
-    }
+    std::cerr << "throw exception" << std::endl;
+    throw Error(ss.str());
```

Comment:
I think this doesn't  work, the original implementation is like this, but it can't stop the program exactly at the line where `CHECK` failed, until the end of the block when destruction is called.

`CHECK` should stop the program immediately, for example:
```c++
{
CHECK(file.is_open());
file.write("xxx");
}
```

if `file` is not open, exception should be throw immediately, not after `file.write`.